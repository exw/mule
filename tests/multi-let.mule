let
  , x = 1
  , y = int.add x 2
in y
