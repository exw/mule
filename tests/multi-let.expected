
Desugared:

	(Let
	 ($1
	  (App
	   ((Fix Record)
	    (Lam
	     ($0
	      (App
	       ((App
	         ((Update (Value x))
	          (App
	           ((App ((Update (Value y)) EmptyRecord))
	            (App
	             ((App ((Var add) (App ((GetField (Lazy x)) (Var $0)))))
	              (Const (Integer 2))))))))
	        (Const (Integer 1))))))))
	  (Let
	   (y (App ((GetField (Strict y)) (Var $1)))
	    (Let (x (App ((GetField (Strict x)) (Var $1))) (Var y)))))))

inferred type:

	(Named (() int))

Runtime term:

	(App
	 ((Lam
	   (1 ()
	    (App
	     ((Lam
	       (1 ()
	        (App ((Lam (1 () (Var 1))) (App ((GetField (Strict x)) (Var 1)))))))
	      (App ((GetField (Strict y)) (Var 0)))))))
	  (App
	   ((Fix Record)
	    (Lam
	     (0 ()
	      (Record
	       ((x (Const (Integer 1)))
	        (y
	         (App
	          ((App ((Prim <fun>) (App ((GetField (Lazy x)) (Var 0)))))
	           (Const (Integer 2)))))))))))))

Evaluated:

	(Const (Integer 3))
